### PR TITLE
Add (disabled) export functionality for MSR

### PR DESCRIPTION
**Pull request by ThanatosGit**
Will be activated if we feel ready for beta test, have a released of the patcher etc.

image
image
image

### CODE REVIEW

**Comment by Miepee** [comment]
if they're the same, whats the reason for needing to differentiate between them?

**Comment by ThanatosGit** [comment]
Different title id, which means the patched romfs needs to be in a different folder for citra and luma. Can be seen in the path created for in the export functions.

Also we may need different exefs patches in the future and then pass the setting to the patcher.

**Comment by henriquegemignani** [comment]
Oh yeah, 3ds has region lock. Wow forgot all about that garbage.

If it's like Switch, then exefs patches uses a hash of the executable so if they're different we can just ship patches for both.

Speaking of, what about exporting both game ids? Thought since we're exporting full pkgs that might be a bit too wasteful.

File: randovania/games/samus_returns/exporter/game_exporter.py [hunk 1]
```diff
@@ -25,10 +51,46 @@ def export_can_be_aborted(self) -> bool:
...
     def _do_export_game(
         self,
         patch_data: dict,
         export_params: GameExportParams,
         progress_update: status_update_lib.ProgressUpdateCallable,
-    ):
-        raise NotImplementedError
+    ) -> None:
+        assert isinstance(export_params, MSRGameExportParams)
+        export_params.output_path.mkdir(parents=True, exist_ok=True)
```
Comment (henriquegemignani) [inline randovania/games/samus_returns/exporter/game_exporter.py:68]: I guess you can't test this function if the import is not included? 🤔 
Maybe keep just the import lines commented out or in a `if False:` block? or maybe even a try/except

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/games/samus_returns/exporter/game_exporter.py [hunk 2]
```diff
@@ -25,10 +51,46 @@ def export_can_be_aborted(self) -> bool:
...
+
+        monitoring.set_tag("msr_mod_category", "pkg")
```
[the remark this answers is not in the document]
Comment (ThanatosGit) [reply randovania/games/samus_returns/exporter/game_exporter.py]: Removed

File: randovania/games/samus_returns/gui/dialog/game_export_dialog.py [hunk 3]
```diff
@@ -1,17 +1,479 @@
...
+    return str(path)
+
+
+def decode_path(s: str | None) -> Path | None:
+    if s is None:
+        return None
+    return Path(s)
+
+
+def get_windows_drives() -> Iterator[tuple[str, str, Path]]:
```
[the remark this answers is not in the document]
Comment (ThanatosGit) [reply randovania/games/samus_returns/gui/dialog/game_export_dialog.py]: Moved it. Had some trouble with mypy.
It ignores `if platform.system [...]` checks and only takes care of `if sys.platform [...]` and also only if it is not in a variable.
Please take some extra look at it. If it is OK, I will squash all the tries / commits.

File: randovania/games/samus_returns/exporter/game_exporter.py (hunk 1 above)
Comment (ThanatosGit) [reply randovania/games/samus_returns/exporter/game_exporter.py:69]: Added a weird test. Will be changed anyways in the next 1-2 weeks.

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**